proto/api.go

```diff
@@ -79,6 +79,7 @@ type Request interface {
 	gogoproto.Message
 	// Header returns the request header.
 	Header() *RequestHeader
+	KVHeader() *KVRequestHeader
```

Comment:
If you're adding `KVHeader()` here, then the interface should be `KVRequest`. I think this should really be split up to be something like:

```
type Request interface {
  gogoproto.Message
  Header() *RequestHeader
  Method() Method
  flags() int
}

type KVRequest interface {
  Request
  KVHeader() *KVRequestHeader
  CreateReply() KVResponse
}
```